### PR TITLE
Fix lower zorder traces behing some gridlines on overlayed suplots

### LINKED ISSUE #7144: zorder property for traces causes gridline to be drawn above traces
The release of plotly.js [v2.35.1](https://github.com/plotly/plotly.js/releases/tag/v2.35.1) solves the problem that the zorder property didn't work correctly with overlayed subplots with multiple traces in each subplot. See #7070 for more details.

With this fix, a new problem was introduced. The gridlines of the subplots are now being influenced by the zorder of the traces, which is an unwanted side effect.

This is demonstrated in [this codepen](https://codepen.io/Djeramon/pen/RwzvXQX).

The blue line is drawn behind the gridlines of the the y2 axis. This should not be the case, as gridlines should always default to be drawn behind traces. Interestingly, the orange line is drawn above the gridlines, even though it is linked to the same yaxis as the blue line and also has a lower zorder than all traces on the y2 axis.

image

cc: @my-tien , @archmoj , @gvwilson

### PR DESCRIPTION
**Pull request by archmoj**
Supersedes #7149.

Here is a minimal code change compared to the rather complex solution proposed in #7149 and avoids several potential side effects catch by tests.

Fixes #7144 thanks to debugging sessions with @emilykl and @gvwilson 

@plotly/plotly_js 
@bpostlethwaite
@stephprobst 
@my-tien

### CODE REVIEW

File: test/image/baselines/zorder-overlayed-subplots-multiple-traces-main-subplot.png
Comment (stephprobst) [inline test/image/baselines/zorder-overlayed-subplots-multiple-traces-main-subplot.png:1]: Isn't the new image incorrect? The overlaying grid lines are gone, but the drawing order of the traces has changed and isn't correct.

- the blue trace 0 with zorder 1 is at the very back
- the orange trace 1 with zorder 2 is at the very front.
- The other two traces with zorder 3 and 4 are in between.

Comment (archmoj) [reply test/image/baselines/zorder-overlayed-subplots-multiple-traces-main-subplot.png:1]: Good catch. Let me fix that.

Comment (archmoj) [reply test/image/baselines/zorder-overlayed-subplots-multiple-traces-main-subplot.png:1]: I had a debugging session with @emilykl and we both agreed that this function is the right place the fix should go into; but still needs extra adjustments.
We will update you on this soon.

**Comment by archmoj** [comment]
@my-tien and @stephprobst Please use [this bundle](https://output.circle-artifacts.com/output/job/07ec9652-70cc-4287-bfb0-c639ea6b2d39/artifacts/0/dist/plotly.min.js) to test this within your system and let us know.

**Comment by archmoj** [comment]
> @my-tien and @stephprobst Please use [this bundle](https://output.circle-artifacts.com/output/job/07ec9652-70cc-4287-bfb0-c639ea6b2d39/artifacts/0/dist/plotly.min.js) to test this within your system and let us know.

@stephprobst @my-tien were you able to test this bundle? https://output.circle-artifacts.com/output/job/07ec9652-70cc-4287-bfb0-c639ea6b2d39/artifacts/0/dist/plotly.min.js

**Comment by stephprobst** [comment]
@archmoj - I'm out of office and won't be able to test the package before the end of the week. Same for My-Tien.

**Comment by archmoj** [comment]
Hi @stephprobst @my-tien 
We are wondering if you tested this PR and zorder works as expected to possibly release a patch.
Please confirm.
Thank you!

**Comment by stephprobst** [comment]
@archmoj: Sorry for the delay in testing. I finally got around to it.

The issue with the gridlines seems to be resolved, but we have a similar situation with the tick labels. Here is a codepen displaying the problem:

https://codepen.io/Djeramon/pen/RwzvXQX

What it currently looks like:
image

What it should look like (tick labels always on top):
image

**Comment by stephprobst** [comment]
@archmoj: I just noticed, that there is no issue tracking this work anymore. Should I create one? Or will you take care of that? Thanks!

**Comment by archmoj** [comment]
The inside tick label edge is tracked in #7227.

**Pull request merged**